concent_api/gatekeeper/views.py

```diff
@@ -74,4 +78,59 @@ def parse_headers(request):
+    assert isinstance(loaded_golem_message, Message)
+
+    # Check if request header contains Concent-Client-Public-Key:
+    if 'HTTP_CONCENT_CLIENT_PUBLIC_KEY' not in request.META:
+        return gatekeeper_access_denied_response('Missing Concent-Client-Public-Key header.')
+
+    # Check ConcentFileTransferToken each field:
+    # -DEADLINE
+    if not isinstance(loaded_golem_message.token_expiration_deadline, int):
+        return gatekeeper_access_denied_response('Wrong type of token_expiration_deadline variable.')
+    current_time = int(datetime.datetime.now().timestamp())
+
+    if current_time > loaded_golem_message.token_expiration_deadline:
+        return gatekeeper_access_denied_response('token_expiration_deadline has passed.')
+
+    # -STORAGE_CLUSTER_ADDRESS
+    if not isinstance(loaded_golem_message.storage_cluster_address, str):
+        return gatekeeper_access_denied_response('Wrong type of storage_cluster_address variable.')
+    cluster_address = URLValidator()
+    try:
+        cluster_address(loaded_golem_message.storage_cluster_address)
+    except ValidationError:
+        return gatekeeper_access_denied_response('storage_cluster_address is not a valid URL.')
+    if loaded_golem_message.storage_cluster_address != settings.STORAGE_CLUSTER_ADDRESS:
+        return gatekeeper_access_denied_response('Given storage_cluster_address is not defined in settings.')
+
+    # -CLIENT_PUBLIC_KEY
+    if not isinstance(loaded_golem_message.authorized_client_public_key, bytes):
+        return gatekeeper_access_denied_response('Wrong type of authorized_client_public_key variable.')
+    client_public_key_base64 = (b64encode(loaded_golem_message.authorized_client_public_key)).decode('ascii')
+    if request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'] != client_public_key_base64:
+        return gatekeeper_access_denied_response('authorized_client_public_key is different then Concent-Client-Public-Key header.')
+
+    # -OPERATION
+    if request.method == 'POST' and loaded_golem_message.operation != 'upload':
+        return gatekeeper_access_denied_response('Wrong operation variable for this request method.')
+    if request.method == 'GET' and loaded_golem_message.operation != 'download':
+        return gatekeeper_access_denied_response('Wrong operation variable for this request method.')
+
+    # -FILES
+    if not all(isinstance(path, str) for path in loaded_golem_message.files):
+        return gatekeeper_access_denied_response('Wrong type of files variable.')
+    if request.method == 'POST':
+        path_to_file = request.get_full_path().partition("/upload-auth/")[2]
+    if request.method == 'GET':
+        path_to_file = request.get_full_path().partition("/download-auth/")[2]
+    if path_to_file not in loaded_golem_message.files:
+        return gatekeeper_access_denied_response('Path to specified file is not listed in files variable.')
+
+    # TMP check if header doesn't contains Concent-Upload-Path or X-Session-ID:
```

Comment:
All `TMP`s should be resolved before sumitting a pull request.

Reply:
I've deleted this part.